components/base/sensorcontrolled/velocities.go

```diff
@@ -123,25 +166,24 @@ func (sb *sensorBase) SetState(ctx context.Context, state []*control.Signal) err
 	defer sb.mu.Unlock()
 	if sb.isPolling() {
 		// if the spin loop is polling, don't call set velocity, immediately return
-		// this allows us to keep the control loop unning without stopping it until
+		// this allows us to keep the control loop running without stopping it until
 		// the resource Close has been called
 		sb.logger.Info("skipping set state call")
 		return nil
 	}
 
-	sb.logger.Info("setting state")
 	linvel := state[0].GetSignalValueAt(0)
-	angvel := state[1].GetSignalValueAt(0)
+	// FIX: multiply angvel by the sign of linvel... why does this work?
+	angvel := (state[1].GetSignalValueAt(0) * sign(linvel))
```

Comment:
different fix for this, if the input linear velocity is negative, change the angular gain block to be negative. makes sense because it's accomplishing the same thing, but still not sure why this is necessary. maybe something with + and - angular direction being flipped when the linear direction is negative?

Reply:
Confirmed the following beahviour on a viam rover just now:
```go
logger.Info("positive y, positive angZ")
	logger.Info("should move forward and turn left")
	if err := viamBase.SetVelocity(context.Background(), r3.Vector{Y: y}, r3.Vector{Z: z}, nil); err != nil {
		logger.Error(err)
	}
logger.Info("positive y, negative angZ")
	logger.Info("should move forward and turn right")
	if err = viamBase.SetVelocity(context.Background(), r3.Vector{Y: y}, r3.Vector{Z: -z}, nil); err != nil {
		logger.Error(err)
	}
logger.Info("negative y, positive angZ")
	logger.Info("should move backward and turn right")
	if err = viamBase.SetVelocity(context.Background(), r3.Vector{Y: -y}, r3.Vector{Z: z}, nil); err != nil {
		logger.Error(err)
	}
logger.Info("negative y, negative angZ")
	logger.Info("should move forward and turn left")
	if err = viamBase.SetVelocity(context.Background(), r3.Vector{Y: -y}, r3.Vector{Z: -z}, nil); err != nil {
		logger.Error(err)
	}
```

Reply:
confirmed!